updater/src/Updater/SyncLabels/Request.purs

```diff
@@ -48,6 +58,68 @@ type Sifted =
   , delete :: Array IssueLabel
   }
 
+listAllLabels :: String -> ExceptT Error Aff Unit
+listAllLabels token = do
+  let
+    allRepos = map { owner: "purescript", repo: _ } purescriptRepos
+      <> map { owner: "purescript-contrib", repo: _ } purescriptContribRepos
+      <> map { owner: "purescript-web", repo: _ } purescriptWebRepos
+      <>
+        map { owner: "purescript-node", repo: _ } purescriptNodeRepos
+
+    getRepoLabel = listLabel <<< (\{ owner, repo } -> { token, owner, repo })
+
+  results <- parTraverse getRepoLabel allRepos
+
+  let
+    handleInsert repo accMap label =
+      { labels: Map.insertWith (<>) label.name [ repo ] accMap.labels
+      , metadata: Map.insertWith Set.union label.name (Set.singleton { description: label.description, color: label.color }) accMap.metadata
+      }
+
+    f accMap { repo, labels } =
+      foldl (handleInsert repo) accMap labels
+
+    labelMap = foldl f { labels: Map.empty, metadata: Map.empty } results
+  liftEffect do
+    log $ i "# of Unique Labels: " (show $ Map.size labelMap.labels)
+    log $ i "Label names: " (show $ sort $ map fst $ (Map.toUnfoldableUnordered labelMap.labels :: Array _))
```

Comment:
Up to you, but I think I'd find this a little easier to parse if we floated these messages up to `let` bindings, like

```purs
  let
    uniqueLabelsCount = Map.size labelMap.labels
    labelNames = sort $ map fst $ (Map.toUnfoldable labelMap.labels :: Array _)
    -- other messages

traverse_ log
  [ "----------"
  , i "# of Unique Labels: " (show uniqueLabelsCount)
  , i "Label Names: " (show labelNames)
  ]
```

(and the same for the later definitions). That way we separate the concerns of calculating the message from printing the message, which is easier for me to follow what's going on.